File: Linked-Lists/Java/array to linked list.cpp
#include <vector>
#include <iostream>
using namespace std;
class Node{
    public:
    int data;
    Node* next;
    public:
    Node(int data1,Node* next1)
    {
        data=data1;
        next=next1;
    }
    public:
    Node(int data1)
    {
        data=data1;
        next=nullptr;
    }
};
 Node* convertarraytolist(vector<int> &arr)
    {
    	Node* head=new Node(arr[0]);
    	Node* mover=head;
    	for(int i=1;i<arr.size();i++)
    	{
    		Node* temp=new Node(arr[i]);
    		mover->next=temp;
    		mover=temp;
		}
		return head;
	}
	int  length(Node * temp)
	{
		int cnt=0;
		while(temp)
		{
			cnt+=1;
			temp=temp->next;
		}
		return cnt;
	}
int main() {
    // Write C++ code here
  
    vector<int>arr={15,5,8,7};
    Node* y=new Node(arr[0]);
    cout<<y->data<<endl;
    Node * head=convertarraytolist(arr);
    cout<<head->data;

    return 0;
}

File: Linked-Lists/Java/array_to_linkedlist.java

public class array_to_linkedlist {


    static class Node{
        int data;
        Node next;

        Node(int data1,Node next1)
        {
            data=data1;
            next=next1;
        }

        Node(int data1)
        {
            data=data1;
            next=null;
        }
    };
    public static Node convertarraytolist(int[] arr)
    {
        Node head=new Node(arr[0]);
        Node mover=head;
        for(int i=1;i<arr.length;i++)
        {
            Node temp=new Node(arr[i]);
            mover.next=temp.next;
            mover=temp;
        }
        return head;
    }

    public static void main(String[] args) {
        int[] arr={15,5,8,7};
        Node y=new Node(arr[0]);
        System.out.println(y.data);
        Node  head=convertarraytolist(arr);
        System.out.println(head.data);

    }



}
